bin/weewx/gettext.py

```diff
@@ -0,0 +1,166 @@
+                # Note: no 'else:' here, because we need the empty dict
+                # page is specified --> get subsection "page"
+                _text_dict = _text_dict.get(page,{})
+                # if the result is not a dict, make it a dict
+                if not isinstance(_text_dict,dict):
+                    _text_dict={page:_text_dict}
+            
+            # if key is not empty, get the value for key
+            if key:
+            
+                if key in _text_dict:
+                    # key is in dict _text_dict --> return value of key
+                    val = _text_dict[key]
+                else:
+                    # otherwise return key as value
+                    val = key
+                
+                return val
+        
+            # if key is empty but page is not return further class instance
+            if page:
+                cheetah_dict={}
+                cheetah_dict_key = 'CheetahGenerator'
+                if "FileGenerator" in self.generator.skin_dict and 'CheetahGenerator' not in self.generator.skin_dict:
+                    cheetah_dict_key = "FileGenerator"
+                if cheetah_dict_key in self.generator.skin_dict:
+                    cheetah_dict = Gettext._get_cheetah_dict(self.generator.skin_dict[cheetah_dict_key],page)
+                return FilesBinder(page,_text_dict,self.generator.skin_dict.get('Labels',{}).get('Generic',{}),cheetah_dict)
+                
+            # if key as well as page are empty
+            return _text_dict
+            
+        return [{'gettext':locale_label}]
+
+    @staticmethod
+    def _get_cheetah_dict(cheetah_dict,page):
+        """ find section page in cheetah_dict recursively """
+    
+        for section in cheetah_dict.sections:
+            subsection = Gettext._get_cheetah_dict(cheetah_dict[section],page)
+            if subsection is not None:
+                return subsection
+        if page in cheetah_dict:
+            return cheetah_dict[page]
+        return None
+
+
+class FilesBinder(object):
+    """ special labels for a specific page to be created by Cheetah """
+
+    def __init__(self,page,lang_files_dict,skin_labels_dict,cheetah_dict):
+        self.page=page
+        self.lang_files_dict=lang_files_dict
+        self.skin_labels_dict=skin_labels_dict
+        self.cheetah_dict=cheetah_dict
+        
+    def __getattr__(self,attr):
+        if attr in self.lang_files_dict:
+            # entry found in localization file
+            return self.lang_files_dict[attr]
+        elif attr in self.cheetah_dict:
+            # entry found [CheetahGenerator] subsection
+            return self.cheetah_dict[attr]
+        elif attr=='nav':
+            # if $gettext(page='xxx').nav not in localization file, try 
+            # nav_xxx in [Labels][[Generic]] section of skin.conf
+            # helpful for extending Belchertown skin
+            if 'nav_'+self.page in self.skin_labels_dict:
```

Comment:
Don't understand why we need this. What's special about the Belchertown skin?

Reply:
So I explain it, but in the end, I remove it, because `FilesBinder` is not needed any more then.

The Belchertown skin (and I guess some other complex skins, too) contains:
```
                #if $page == "graphs"
                <h1>$obs.label.graphs_page_header</h1>
                #else if $page == "records"
                <h1>$obs.label.records_page_header</h1>
                #else if $page == "reports"
                <h1>$obs.label.reports_page_header</h1>
                #else if $page == "about"
                <h1>$obs.label.about_page_header</h1>
                #else if $page == "imprint"
                <h1>$obs.label.imprint_page_header</h1>
                #else if $page == "gdpr"
                <h1>$obs.label.gdpr_page_header</h1>
                #end if
```
If you want to add a new page, you have to write another `#else if`.

With the new `$gettext` feature that can be replaced by a single line:
```
<h1>$gettext("page_header",$page)</h1>
```
No need to change or add anything when adding a new page any more.

Now, the values are found in `[Labels][[Generic]]` instead of `[Texts]`, and they include the page's name. The code you asked about converted `page_header` and the value of `$page` to the key to look for in `[Labels][[Generic]`. 

But there is an alternative. You can move all those entries from `[Labels][[Generic]]` to `[Texts]`, and then you do not need special handling of those tags any more.

So I removed that. And I did point 6 of your comment with that as well.